### PR TITLE
Initialize variables to avoid the warning message when compiling

### LINKED ISSUE #751: GSI for RRFS does not build without warning prints
NCO has been getting stricter about codes compiling cleanly, without warnings.  I see warning prints, but limited to a single routine (src/enkf/observer_fv3reg.f90), that are of this type:

```
/lfs/h2/emc/lam/noscrub/Matthew.Pyle/rrfs-workflow_warnings/sorc/gsi/src/enkf/observer_fv3reg.f90(24): warning #6843: A dummy argument with an explicit INTENT(OUT) declaration is not given an explicit value.   [IX]
subroutine setup_linhx(rlat, rlon, time, ix, delx, ixp, delxp, iy, dely,  &
-----------------------------------------^
```

Would like to see this item cleaned up well ahead of code delivery for the RRFS - hopefully a straightforward item to correct.

### PR DESCRIPTION
**Pull request by ShunLiu-NOAA**
**Description**

This PR is to initialize variables in observer_fv3reg.f90 to avoid the warning message when compiling. 
The change will update regional EnKF excutable. However, the current regression test can't identify the changes in regional EnKF mode. The difference between before and after this update is that no warning message appears when compiling regional EnKF. 

This PR is to address https://github.com/NOAA-EMC/GSI/issues/751

**Type of change**

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

  
**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

File: src/enkf/observer_fv3reg.f90 [hunk 1]
```diff
@@ -54,6 +54,18 @@ subroutine setup_linhx(rlat, rlon, time, ix, delx, ixp, delxp, iy, dely,  &
   real(r_single)                                   ,intent(in   ) :: time         ! observation time relative to middle of window
   integer(i_kind), intent(out) :: ix, iy, it, ixp, iyp, itp
   real(r_kind), intent(out) :: delx, dely, delxp, delyp, delt, deltp
+  ix=0.0
```
Comment (RussTreadon-NOAA) [inline src/enkf/observer_fv3reg.f90]: Specify precision `_r_kind` or use the constant `zero`.

File: src/enkf/observer_fv3reg.f90 [hunk 2]
```diff
@@ -110,6 +122,7 @@ subroutine calc_linhx(hx, dens, dhx_dx, hxpert, hx_ens, &
   type(raggedarr)                                  ,intent(inout) :: hxpert       ! interpolated background
   real(r_single)                                   ,intent(  out) :: hx_ens       ! H (x_ens)
   integer(i_kind) i,j
+  hx_ens=0.0
```
Comment (RussTreadon-NOAA) [inline src/enkf/observer_fv3reg.f90]: Same comment as above

File: src/enkf/observer_fv3reg.f90 [hunk 3]
```diff
@@ -155,6 +168,7 @@ subroutine calc_linhx_modens(hx, dhx_dx, hxpert, hx_ens, vscale)
   real(r_single)                                   ,intent(  out) :: hx_ens(neigv)! H (x_ens)
   real(r_double),dimension(neigv,nlevs+1)          ,intent(in   ) :: vscale       ! vertical scaling (for modulated ens)
   integer(i_kind) i
+  hx_ens=0.0
```
Comment (RussTreadon-NOAA) [inline src/enkf/observer_fv3reg.f90]: Same comment as above

**Comment by RussTreadon-NOAA** [comment]
We can use the regional enkf ctest in PR #750 to test the changes in this PR, right?

**Comment by hu5970** [comment]
@RussTreadon-NOAA Yes, we can use the regional EnKF case from PR #750. But need to turn on the case manually.

**Comment by RussTreadon-NOAA** [comment]
> @RussTreadon-NOAA Yes, we can use the regional EnKF case from PR #750. But need to turn on the case manually.

Good.  So we can test if initializing variables to zero has any impact ... at least in the context of the regional enkf ctest.

File: src/enkf/observer_fv3reg.f90 (hunk 1 above)
Comment (ShunLiu-NOAA) [reply src/enkf/observer_fv3reg.f90]: code is updated to use izero and zero

**Review by RussTreadon-NOAA — APPROVED** [review]
Thank you @ShunLiu-NOAA for making these changes.  Looks good to me.  Initializing variables to zero should not alter results, but it would be good to confirm this via the regional enkf ctest in PR #750 or an offline test.

Approve pending _no impact_ result from ctest or offline test.

**Comment by ShunLiu-NOAA** [comment]
@RussTreadon-NOAA Thank you for reviewing the changes. I will start an offline test.

**Review by TingLei-NOAA — APPROVED** [review]

**Comment by TingLei-NOAA** [comment]
Those added initialization  are in "dummy " subroutines ( as shown by those lines 
```
write(6,*)'this is a dummy subroutine, running this means something wrong ,stop'
   call stop2(555)
```
So, in my opinion, no need to do extra test to see if the will have impacts on the results if we are sure there are no changes outside of those dummy subroutines ( while the whole piece (observer_fv3reg.f90) is for those dummy subroutine)

**Comment by RussTreadon-NOAA** [comment]
@ShunLiu-NOAA , what is the status of this PR?

**Comment by RussTreadon-NOAA** [comment]
@ShunLiu-NOAA , what is the status of this PR?

**Comment by ShunLiu-NOAA** [comment]
Test project /scratch1/NCEPDEV/da/Shun.Liu/gsi_regression/GSI/build
    Start 1: global_4denvar
    Start 2: rtma
    Start 3: rrfs_3denvar_rdasens
    Start 4: hafs_4denvar_glbens
    Start 5: hafs_3denvar_hybens
    Start 6: global_enkf
1/6 Test #3: rrfs_3denvar_rdasens .............   Passed  859.46 sec
2/6 Test #6: global_enkf ......................   Passed  1298.20 sec
3/6 Test #5: hafs_3denvar_hybens ..............   Passed  1403.31 sec
4/6 Test #2: rtma .............................   Passed  2170.92 sec
5/6 Test #4: hafs_4denvar_glbens ..............   Passed  2186.12 sec
6/6 Test #1: global_4denvar ...................   Passed  2886.83 sec

100% tests passed, 0 tests failed out of 6

image

**Comment by ShunLiu-NOAA** [comment]
@hu5970 ctest on WCOSS and HERA passed. The PR is ready to merge into _develop_.

**Comment by RussTreadon-NOAA** [comment]
Thank you @ShunLiu-NOAA for running ctests.  My previous approval of this PR remains.  These changes address a [WCOSS Implementation Standard](https://www.nco.ncep.noaa.gov/idsb/implementation_standards/ImplementationStandards.v11.0.0.pdf) 

> 5. Code must compile without errors or warnings. Errors and warnings may not be suppressed, and the compiler warning level ("-W" options) must be at least the default one.

@hu5970 , is it possible for you to run the regional enkf ctest for the sake of completeness?   Initializing variable to `izero` or `zero` should have no impact on regional enkf results.  It would be nice to confirm this.

**Comment by ShunLiu-NOAA** [comment]
@RussTreadon-NOAA Thank you. I will coordinate with @hu5970 for a regional enkf test.

**Comment by hu5970** [comment]
I run a regression test suite with the regional EnKF case on and there are the results:

```
Test project /scratch1/BMC/wrfruc/mhu/code/gsi/GSI/build
    Start 1: global_4denvar
    Start 2: rtma
    Start 3: rrfs_3denvar_rdasens
    Start 4: hafs_4denvar_glbens
    Start 5: hafs_3denvar_hybens
    Start 6: rrfs_enkf_conv

1/6 Test #6: rrfs_enkf_conv ...................   Passed  303.82 sec
2/6 Test #3: rrfs_3denvar_rdasens .............   Passed  492.80 sec
3/6 Test #2: rtma .............................   Passed  1093.14 sec
4/6 Test #5: hafs_3denvar_hybens ..............   Passed  1109.30 sec
5/6 Test #4: hafs_4denvar_glbens ..............   Passed  1349.62 sec
6/6 Test #1: global_4denvar ...................   Passed  2028.61 sec

100% tests passed, 0 tests failed out of 6

Total Test time (real) = 2028.63 sec
```

**Comment by hu5970** [comment]
I also updated the ctest instruction on how to run ctest with regional enkf case.

**Pull request merged**